DigitalLearningSolutions.Web/Controllers/SuperAdmin/Centres/CentresController.cs

```diff
@@ -480,30 +480,38 @@ public IActionResult EditContractInfo(int centreId , int? day, int? month, int?
 
         [Route("SuperAdmin/Centres/{centreId=0:int}/EditContractInfo")]
         [HttpPost]
-        public IActionResult EditContractInfo(ContractTypeViewModel contractTypeViewModel,int? day,int? month,int? year)
+        public IActionResult EditContractInfo(ContractTypeViewModel contractTypeViewModel, int? day, int? month, int? year)
         {
-            if (day != 0 | month != 0 | year != 0)
+            if ((day != 0 && day != null) | (month != 0 && month != null) | (year != 0 && year != null))

```

Comment:
(day.GetValueOrDefault() != 0) || (month.GetValueOrDefault() !=0) || (year.GetValueOrDefault() != 0)

use GetValueOrDefault() because it is faster and cleaner.
Also, we do not want the whole expression to get evaluated before going into the if condition so use double pipes so when the condition is true, the rest of the expression is not evaluated.